egs/wsj/s5/steps/resolve_ctm_overlaps.py

```diff
@@ -0,0 +1,319 @@
+#! /usr/bin/env python
+
+# Copyright 2014  Johns Hopkins University (Authors: Daniel Povey)
+#           2014  Vijayaditya Peddinti
+#           2016  Vimal Manohar
+# Apache 2.0.
+
+"""
+Script to combine ctms with overlapping segments.
+The current approach is very simple. It ignores the words,
+which are hypothesized in the half of the overlapped region
+that is closer to the utterance boundary.
+So if there are two segments
+in the region 0s to 30s and 25s to 55s, with overlap of 5s,
+the last 2.5s of the first utterance i.e. from 27.5s to 30s is truncated
+and the first 2.5s of the second utterance i.e. from 25s to 27.s is truncated.
+"""
+
+from __future__ import print_function
+import argparse
+import collections
+import logging
+
+from collections import defaultdict
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+handler = logging.StreamHandler()
+handler.setLevel(logging.INFO)
+formatter = logging.Formatter(
+    '%(asctime)s [%(pathname)s:%(lineno)s - '
+    '%(funcName)s - %(levelname)s ] %(message)s')
+handler.setFormatter(formatter)
+logger.addHandler(handler)
+
+
+def get_args():
+    """gets command line arguments"""
+
+    usage = """ Python script to resolve overlaps in ctms """
+    parser = argparse.ArgumentParser(usage)
+    parser.add_argument('segments', type=argparse.FileType('r'),
+                        help='use segments to resolve overlaps')
+    parser.add_argument('ctm_in', type=argparse.FileType('r'),
+                        help='input_ctm_file')
+    parser.add_argument('ctm_out', type=argparse.FileType('w'),
+                        help='output_ctm_file')
+    parser.add_argument('--verbose', type=int, default=0,
+                        help="Higher value for more verbose logging.")
+    args = parser.parse_args()
+
+    if args.verbose > 2:
+        logger.setLevel(logging.DEBUG)
+        handler.setLevel(logging.DEBUG)
+
+    return args
+
+
+def read_segments(segments_file):
```

Comment:
I am missing a lot of details, so this might not be the right idea, but does it make sense to use the smart_open() magic in these functions? This would allow stdin/out piping.